### PR TITLE
#12

### PR DESCRIPTION
**Pull request by anuratb**
Solves #12

### CODE REVIEW

**Comment by pshishod2645** [comment]
Hi please add a precise description of what all changes you made, and attach the screenshots. Ping me/ @TheLethalCode  on slack once you do this, and one of us will review your pull request.

**Comment by anuratb** [comment]
[report12.pdf](https://github.com/pshishod2645/GOC-Website/files/5693723/report12.pdf)

**Comment by TheLethalCode** [comment]
hey @anuratb. Just some guidelines for PR. The PR title should give the problem you handled, and in the description, you give a concise overview of how you did it. Here, you gave out a document, which is not the right way to go about it, as it defeats the purpose. You should attach the screenshots and other related entities directly in the comment.

That aside, on the changes you made,
1) The fields for the "contents" to be much bigger in size and not a single line field.
2) Note that not all those who are there for shortlisting rounds are selected to attend the interview. Ask whether they are shortlisted, and if yes, give them the fields for the interview.
3) BlogId is not filled up by the author, it is generated in the backend.
4) Why is add company an option? Isn't it mandatory?
5) Is the final submit button for the form "Add Blog"? if so, could you differentiate it from the rest of the buttons?

File: goc/app.py [hunk 1]
```diff
@@ -45,17 +54,86 @@ def blog():
...
+        blog_form.tags.append_entry()
+        if(len(blog_form.interview.interview_rounds)>0):
+            return render_template('blogform.j2',blog_form = blog_form,shortlisted = 1)
+        else:
+            return render_template('blogform.j2',blog_form = blog_form,shortlisted = 0)
+    if(blog_form.validate_on_submit()):       
+        #First make all tags unique
+        tags = [str(x) for x in set(blog_form.tags)]   
+        author = Author(name = str(blog_form.author))                   
+        db.session.add(author)
```
Comment (TheLethalCode) [inline goc/app.py:95]: You should put the entire db changes inside try except block to handle unexpected errors. If while adding data to the db raises an error, you can return an error through the except block

File: goc/models.py [hunk 2]
```diff
@@ -29,7 +33,7 @@ def __repr__(self):
 class Author(db.Model):
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(20), nullable=False)
-    blogs = db.relationship('Blog', backref='author', lazy=True)
+    blogs = db.relationship('Blog', backref='authorr', lazy=True)
```
Comment (TheLethalCode) [inline goc/models.py:36]: Is this intended?

**Comment by TheLethalCode** [comment]
You need not have a separate commit for each file change. You can just combine them into a single commit to reduce the clutter. 

@pshishod2645 review it once.

File: goc/models.py (hunk 2 above)
Comment (anuratb) [reply goc/models.py:36]: Without this errors are coming .The names under backref and the attribute author in blog defining a relationship need to be different

File: goc/app.py (hunk 1 above)
Comment (anuratb) [reply goc/app.py]: Done it,Thanks

**Pull request closed without merging**